apps/server/src/modules/school/repo/mikro-orm/school.repo.ts

```diff
@@ -39,6 +39,18 @@ export class SchoolMikroOrmRepo implements SchoolRepo {
 		return school;
 	}
 
+	public async getSchoolsBySystemIds(systemIds: string[]): Promise<School[]> {
+		const entities = await this.em.find(
+			SchoolEntity,
+			{ systems: { $in: systemIds } },
```

Comment:
Ich frage mich, ob mikrorom strings ins ObjectId umwandelt?

Reply:
Ist das ne allgemeine Frage oder speziell auf die Stelle bezogen? Normalerweise kann man IDs ja als Strings an MikroOrm übergeben, siehe z.B.: https://mikro-orm.io/docs/5.9/entity-manager#conditions-object-filterqueryt Ich erwarte hier kein anderes Verhalten und es funktioniert auch.

Reply:
Ne speziell auf diese Stelle im db.schools.systems: [ObjectId("abc")] also sind die ObjectIds gespeichert. und insuche mit $in: ["abc"] suchst du nach strings. Es wird zu kein Ergebnis führen. Ich was das ich im TypeORM musste ich so was machen wie const systemId = sustems.map(sustem => new ObjectId(system.id))

Reply:
Es führt wie gesagt zu einem Ergebnis. Das ist doch auch die gleiche Situation, wie wenn du direkt mit der ID suchst. Da steht in der Datenbank doch auch ne ObjectId, aber du übergibst nen String.

Ich hab gerade mal im Code von MikroOrm geschaut. Ich nehme an, dass an dieser Stelle IDs in String-form zu ObjectIds umgewandelt werden: https://github.com/mikro-orm/mikro-orm/blob/9eb9f90d1b63cfea83cbdb4d9ef1bb37d9b966fb/packages/mongodb/src/MongoDriver.ts#L320 Aber ist nicht gerade leichtverständlich...

Reply:
Das ist übrigens nicht der $in-Operator von MongoDB, also nicht direkt eine Mongo-Query, sondern der von MikroOrm: https://mikro-orm.io/docs/5.9/query-conditions

Vielleicht ist das bei TypeORM anders? Dass man da direkt eine Mongo-Query übergeben kann?